llvm/lib/SYCLLowerIR/LowerESIMD.cpp

```diff
@@ -252,6 +252,15 @@ class ESIMDIntrinDescTable {
         {"flat_write", {"svm.scatter", {ai1(3), a(2), a(0), a(1)}}},
         {"flat_write4",
          {"svm.scatter4.scaled", {ai1(2), t(2), c16(0), c64(0), a(0), a(1)}}},
+
+        // surface index-based gather/scatter:
+        // num blocks, scale, surface index, global offset, elem offsets
+        {"surf_read", {"gather.scaled2", {t(3), c16(0), aSI(1), a(2), a(3)}}},
```

Comment:
Why not map to gather.scaled? It may need general predicate support.

Reply:
Good question. I'm actually following advice from the BE team (@aus-intel) who recommended to use this one for now. When predication is needed, we can add this w/o breaking user code.